rego/rego.go

```diff
@@ -1851,6 +1878,116 @@ func (r *Rego) eval(ctx context.Context, ectx *EvalContext) (ResultSet, error) {
 	return rs, nil
 }
 
+func (r *Rego) evalWasm(ctx context.Context, ectx *EvalContext) (ResultSet, error) {
+
+	var input *interface{}
+	if ectx.parsedInput != nil {
+		i, err := ast.JSON(ectx.parsedInput)
+		if err != nil {
+			return nil, err
+		}
+		input = &i
+	}
+
+	result, err := r.opa.Eval(ctx, opa.EvalOpts{Metrics: r.metrics, Input: input})
+	if err != nil {
+		return nil, err
+	}
+
+	parsed, err := ast.ParseTerm(string(result.Result))
+	if err != nil {
+		return nil, err
+	}
+
+	resultSet, ok := parsed.Value.(ast.Set)
+	if !ok {
+		return nil, fmt.Errorf("illegal result type")
+	}
+
+	if resultSet.Len() == 0 {
+		return nil, nil
+	}
+
+	if resultSet.Len() > 1 {
+		return nil, fmt.Errorf("illegal result type")
+	}
+
+	var obj ast.Object
+	err = resultSet.Iter(func(term *ast.Term) error {
+		obj, ok = term.Value.(ast.Object)
+		if !ok {
+			return fmt.Errorf("illegal result type")
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	qr := topdown.QueryResult{}
+
+	err = obj.Iter(func(k, v *ast.Term) error {
+		kvt := ast.VarTerm(string(k.Value.(ast.String)))
+		qr[kvt.Value.(ast.Var)] = v
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
```

Comment:
[nit] I guess we could go with `(*Object).Foreach(func(*ast.Term, *ast.Term))` instead, since there are no errors,

```suggestion
	obj.Foreach(func(k, v *ast.Term) {
		kvt := ast.VarTerm(string(k.Value.(ast.String)))
		qr[kvt.Value.(ast.Var)] = v
	})
```

Reply:
Done